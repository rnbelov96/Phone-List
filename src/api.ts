import axios from 'axios';

export const createApi = () => {
  const api = axios.create({
    baseURL: 'http://jsonplaceholder.typicode.com/photos',
    timeout: 5000,
  });

  return api;
};
