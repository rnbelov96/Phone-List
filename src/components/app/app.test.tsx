import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { createApi } from '../../api';
import App from './app';

const api = createApi();

const mockStore = configureStore([thunk.withExtraArgument(api)]);

describe('App', () => {
  test('should render loading screen', () => {
    const store = mockStore({
      app: {
        modalWindowUrl: null,
        isDataLoading: true,
        currentPage: 1,
        idFilter: 0,
      },
      data: {
        photoList: [
          {
            albumId: 1,
            id: 1,
            thumbnailUrl: 'some url',
            title: 'Some title',
            url: 'some url',
          },
          {
            albumId: 2,
            id: 2,
            thumbnailUrl: 'some url',
            title: 'Some title',
            url: 'some url',
          },
          {
            albumId: 3,
            id: 3,
            thumbnailUrl: 'some url',
            title: 'Some title',
            url: 'some url',
          },
        ],
      },
    });
    const tree = renderer
      .create(
        <Provider store={store}>
          <App />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render main screen', () => {
    const store = mockStore({
      app: {
        modalWindowUrl: null,
        isDataLoading: false,
        currentPage: 1,
        idFilter: 0,
      },
      data: {
        photoList: [
          {
            albumId: 1,
            id: 1,
            thumbnailUrl: 'some url',
            title: 'Some title',
            url: 'some url',
          },
          {
            albumId: 1,
            id: 1,
            thumbnailUrl: 'some url',
            title: 'Some title',
            url: 'some url',
          },
          {
            albumId: 1,
            id: 1,
            thumbnailUrl: 'some url',
            title: 'Some title',
            url: 'some url',
          },
        ],
      },
    });
    const tree = renderer
      .create(
        <Provider store={store}>
          <App />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should render modal window', () => {
    const store = mockStore({
      app: {
        modalWindowUrl: 'some url',
        isDataLoading: false,
        currentPage: 1,
        idFilter: 0,
      },
      data: {
        photoList: [
          {
            albumId: 1,
            id: 1,
            thumbnailUrl: 'some url',
            title: 'Some title',
            url: 'some url',
          },
          {
            albumId: 1,
            id: 1,
            thumbnailUrl: 'some url',
            title: 'Some title',
            url: 'some url',
          },
          {
            albumId: 1,
            id: 1,
            thumbnailUrl: 'some url',
            title: 'Some title',
            url: 'some url',
          },
        ],
      },
    });
    const tree = renderer
      .create(
        <Provider store={store}>
          <App />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
