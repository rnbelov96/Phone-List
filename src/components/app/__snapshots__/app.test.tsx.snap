// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render loading screen 1`] = `
<div
  className="container"
>
  <section
    className="loading"
  >
    <p
      className="loading__text"
    >
      Loading in progress...
    </p>
    <div
      className="lds-ring mx-auto loading__circle"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
  </section>
</div>
`;

exports[`App should render main screen 1`] = `
<div
  className="container"
>
  <section
    className="app"
  >
    <section
      className="filter"
    >
      <p
        className="filter__text"
      >
        Filter by id:
      </p>
      <select
        className="form-select filter__select"
        onChange={[Function]}
      >
        <option
          value={0}
        >
          All
        </option>
        <option
          value={1}
        >
          1
        </option>
      </select>
    </section>
    <section
      className="list"
    >
      <div
        className="card list__item"
      >
        <img
          alt="small square"
          className="card-img-top list__item-img"
          onClick={[Function]}
          onLoad={[Function]}
          src="some url"
          style={
            Object {
              "display": "none",
              "minHeight": "150px",
            }
          }
        />
        <div
          className="lds-ring mx-auto list__item-circle"
          style={
            Object {
              "display": "block",
              "minHeight": "150px",
            }
          }
        >
          <div />
          <div />
          <div />
          <div />
        </div>
        <div
          className="card-body list__item-content"
        >
          <h5
            className="card-title list__item-title"
          >
            Some title
          </h5>
          <button
            className="btn btn-danger list__item-btn"
            onClick={[Function]}
            type="button"
          >
            Delete Photo
          </button>
        </div>
      </div>
      <div
        className="card list__item"
      >
        <img
          alt="small square"
          className="card-img-top list__item-img"
          onClick={[Function]}
          onLoad={[Function]}
          src="some url"
          style={
            Object {
              "display": "none",
              "minHeight": "150px",
            }
          }
        />
        <div
          className="lds-ring mx-auto list__item-circle"
          style={
            Object {
              "display": "block",
              "minHeight": "150px",
            }
          }
        >
          <div />
          <div />
          <div />
          <div />
        </div>
        <div
          className="card-body list__item-content"
        >
          <h5
            className="card-title list__item-title"
          >
            Some title
          </h5>
          <button
            className="btn btn-danger list__item-btn"
            onClick={[Function]}
            type="button"
          >
            Delete Photo
          </button>
        </div>
      </div>
      <div
        className="card list__item"
      >
        <img
          alt="small square"
          className="card-img-top list__item-img"
          onClick={[Function]}
          onLoad={[Function]}
          src="some url"
          style={
            Object {
              "display": "none",
              "minHeight": "150px",
            }
          }
        />
        <div
          className="lds-ring mx-auto list__item-circle"
          style={
            Object {
              "display": "block",
              "minHeight": "150px",
            }
          }
        >
          <div />
          <div />
          <div />
          <div />
        </div>
        <div
          className="card-body list__item-content"
        >
          <h5
            className="card-title list__item-title"
          >
            Some title
          </h5>
          <button
            className="btn btn-danger list__item-btn"
            onClick={[Function]}
            type="button"
          >
            Delete Photo
          </button>
        </div>
      </div>
    </section>
    <div
      className="page-list"
    >
      <button
        className="page-list__button page-list__button_current "
        onClick={[Function]}
        type="button"
      >
        1
      </button>
    </div>
  </section>
</div>
`;

exports[`App should render modal window 1`] = `
<div
  className="container"
>
  <section
    className="app"
  >
    <div
      className="modal-window"
    >
      <div
        className="modal-window__center-wrapper"
        onClick={[Function]}
      >
        <img
          alt="big square"
          onLoad={[Function]}
          src="some url"
          style={
            Object {
              "display": "none",
            }
          }
        />
        <div
          className="lds-ring mx-auto"
          style={
            Object {
              "display": "inline-block",
            }
          }
        >
          <div />
          <div />
          <div />
          <div />
        </div>
      </div>
    </div>
    <section
      className="filter"
    >
      <p
        className="filter__text"
      >
        Filter by id:
      </p>
      <select
        className="form-select filter__select"
        onChange={[Function]}
      >
        <option
          value={0}
        >
          All
        </option>
        <option
          value={1}
        >
          1
        </option>
      </select>
    </section>
    <section
      className="list"
    >
      <div
        className="card list__item"
      >
        <img
          alt="small square"
          className="card-img-top list__item-img"
          onClick={[Function]}
          onLoad={[Function]}
          src="some url"
          style={
            Object {
              "display": "none",
              "minHeight": "150px",
            }
          }
        />
        <div
          className="lds-ring mx-auto list__item-circle"
          style={
            Object {
              "display": "block",
              "minHeight": "150px",
            }
          }
        >
          <div />
          <div />
          <div />
          <div />
        </div>
        <div
          className="card-body list__item-content"
        >
          <h5
            className="card-title list__item-title"
          >
            Some title
          </h5>
          <button
            className="btn btn-danger list__item-btn"
            onClick={[Function]}
            type="button"
          >
            Delete Photo
          </button>
        </div>
      </div>
      <div
        className="card list__item"
      >
        <img
          alt="small square"
          className="card-img-top list__item-img"
          onClick={[Function]}
          onLoad={[Function]}
          src="some url"
          style={
            Object {
              "display": "none",
              "minHeight": "150px",
            }
          }
        />
        <div
          className="lds-ring mx-auto list__item-circle"
          style={
            Object {
              "display": "block",
              "minHeight": "150px",
            }
          }
        >
          <div />
          <div />
          <div />
          <div />
        </div>
        <div
          className="card-body list__item-content"
        >
          <h5
            className="card-title list__item-title"
          >
            Some title
          </h5>
          <button
            className="btn btn-danger list__item-btn"
            onClick={[Function]}
            type="button"
          >
            Delete Photo
          </button>
        </div>
      </div>
      <div
        className="card list__item"
      >
        <img
          alt="small square"
          className="card-img-top list__item-img"
          onClick={[Function]}
          onLoad={[Function]}
          src="some url"
          style={
            Object {
              "display": "none",
              "minHeight": "150px",
            }
          }
        />
        <div
          className="lds-ring mx-auto list__item-circle"
          style={
            Object {
              "display": "block",
              "minHeight": "150px",
            }
          }
        >
          <div />
          <div />
          <div />
          <div />
        </div>
        <div
          className="card-body list__item-content"
        >
          <h5
            className="card-title list__item-title"
          >
            Some title
          </h5>
          <button
            className="btn btn-danger list__item-btn"
            onClick={[Function]}
            type="button"
          >
            Delete Photo
          </button>
        </div>
      </div>
    </section>
    <div
      className="page-list"
    >
      <button
        className="page-list__button page-list__button_current "
        onClick={[Function]}
        type="button"
      >
        1
      </button>
    </div>
  </section>
</div>
`;
